haystack/nodes/prompt/prompt_node.py

```diff
@@ -187,65 +191,56 @@ def prompt(self, prompt_template: Optional[Union[str, PromptTemplate]], *args, *
                 results.extend(output)
         return results
 
-    def add_prompt_template(self, prompt_template: PromptTemplate) -> None:
-        """
-        Adds a prompt template to the list of supported prompt templates.
-        :param prompt_template: The PromptTemplate object to be added.
-        :return: None
-        """
-        if prompt_template.name in self.prompt_templates:
-            raise ValueError(
-                f"Prompt template {prompt_template.name} already exists. "
-                f"Select a different name for this prompt template."
-            )
-
-        self.prompt_templates[prompt_template.name] = prompt_template  # type: ignore
-
-    def remove_prompt_template(self, prompt_template: str) -> PromptTemplate:
-        """
-        Removes a prompt template from the list of supported prompt templates.
-        :param prompt_template: Name of the prompt template to be removed.
-        :return: PromptTemplate object that was removed.
-        """
-        if prompt_template not in self.prompt_templates:
-            raise ValueError(f"Prompt template {prompt_template} does not exist")
-
-        return self.prompt_templates.pop(prompt_template)
+    @property
+    def default_prompt_template(self):
+        return self._default_prompt_template
 
-    def set_default_prompt_template(self, prompt_template: Union[str, PromptTemplate]) -> "PromptNode":
+    @default_prompt_template.setter
+    def default_prompt_template(self, prompt_template: Union[str, PromptTemplate, None]):
         """
         Sets the default prompt template for the node.
         :param prompt_template: The prompt template to be set as default.
         :return: The current PromptNode object.
         """
-        if not self.is_supported_template(prompt_template):
-            raise ValueError(f"{prompt_template} not supported, select one of: {self.get_prompt_template_names()}")
-
-        self.default_prompt_template = prompt_template
-        return self
-
-    def get_prompt_templates(self) -> List[PromptTemplate]:
-        """
-        Returns the list of supported prompt templates.
-        :return: List of supported prompt templates.
+        self._default_prompt_template = self.get_prompt_template(prompt_template)
+
+    @tenacity.retry(
+        reraise=True,
+        retry=tenacity.retry_if_exception_type((HTTPError, RequestException, JSONDecodeError)),
+        wait=tenacity.wait_exponential(multiplier=PROMPTHUB_BACKOFF),
+        stop=tenacity.stop_after_attempt(PROMPTHUB_MAX_RETRIES),
+    )
+    def _prompt_template_exists_in_hub(self, name):
         """
-        return list(self.prompt_templates.values())
+        Returns True if the prompt exists in the Hub, False otherwise.
 
-    def get_prompt_template_names(self) -> List[str]:
+        Might raise HTTPError.
         """
-        Returns the list of supported prompt template names.
-        :return: List of supported prompt template names.
+        try:
```

Comment:
@ZanSara could we perhaps concentrate all the logic of fetching in one place (i.e. `_get_prompt_template_from_hub`) with all the proper error handling and then use that method here to conclude if we should return True or False?